### ISSUE TITLE
eldoc-mode too slow?

### ISSUE THREAD

**Issue by greghendershott** [greghendershott/racket-mode#199]
Although #126 fixed a contract violation that @BitPuffin had reported, on `#racket` today @BitPuffin mentioned that there are times they see "Waiting for Racket..." frequently. It sounds like this is related to eldoc-mode.

**Comment by greghendershott** [comment]
See also #181. Although #181 is about Emacs 25+ enabling eldoc by default, that means any issues related to eldoc issues will soon start coming up more frequently.

**Comment by greghendershott** [comment]
A couple ideas:

1. Don't automatically start a Racket REPL just for eldoc. If not already running, do nothing. CIDER seems to do this and it seems reasonable.

2. Just generally, eldoc seems of limited utility with Racket (which is why I don't bother using it, and it doesn't get dogfooded):
    - There's no official way to get the "signature" of a function in Racket. I go spelunking for the Typed Racket type, or, contract, or, at least just the arg names in the definition.
    - Quite a few definitions are invisible to this because they're defined in `#%kernel`. So you get no eldoc info for basic forms like `if`, `cond`, and so on.
    - Even when a signature is available, 9 times out of 10 I find myself needing/wanting to see the full (usually excellent) Racket documentation for it, anyway.

    In other words, eldoc works for a sweet spot: Simple functions where all you need to know is the name to tell (or remind) you what the args mean and their order.  I'm not sure that sweet spot is very big, for Racket.  So my experience is that eldoc is a lot of flashing stuff in the echo area that's rarely useful, and just a distraction. Perhaps racket-mode shouldn't set an `eldoc-documentation-function`, at all?!  Especially in an Emacs 25+ world where global eldoc mode is enabled by default.

**Comment by divyaranjan1905** [comment]
@greghendershott Hello, it's been 8 years I know but because I use Racket and Emacs, so here's a few things. I agree with you that `eldoc` on the echo just to see the structure of the s-expression doesn't really have much. But, one can do much more with `eldoc`. Below is a screenshot of a Rust source file where I can get information about anything (keyword or expression) just by hovering it:

![pic-selected-240827-0346-37](https://github.com/user-attachments/assets/6bbdb9ef-d735-4d82-a839-62958190f166)

This is done through `eldoc` which is getting the necessary stuff from `rust-analyzer` which is being managed through `eglot` and then put into a nice hovering box using [`eldoc-box`](https://github.com/casouri/eldoc-box). I think _this_ can be put to good use alongside the excellent documentation that Racket already holds. Moreover, this also works for one's own local definitions:

![pic-selected-240827-0351-26](https://github.com/user-attachments/assets/8224e2eb-2730-412b-ba74-b0fb6e20c976)

This works similarly with Haskell using `haskell-mode` as well. I am not really sure how one might integrate this with `racket-mode`, but if you can share your thoughts on this and maybe guide a bit on how to make `eldoc` talk to Racket documentation I could try for implementing this.

**Comment by divyaranjan1905** [comment]
I just realized `racket-xp-describe` does exactly the same thing in a separate buffer, all that needs tobe done is channel it to the `eldoc-box` childframe. I don't think anything else needs to be done, thank you.
